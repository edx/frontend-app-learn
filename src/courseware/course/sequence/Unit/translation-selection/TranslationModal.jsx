import React from 'react';
import PropTypes from 'prop-types';

import { useIntl } from '@edx/frontend-platform/i18n';
import {
  StandardModal,
  ActionRow,
  Button,
  Icon,
  ListBox,
  ListBoxOption,
} from '@edx/paragon';
import { Check } from '@edx/paragon/icons';

import useTranslationModal from './useTranslationModal';
import { languages } from './useSelectLanguage';
import messages, { languageMessages } from './messages';

import './TranslationModal.scss';

const TranslationModal = ({
  isOpen, close, selectedLanguage, setSelectedLanguage,
}) => {
  const { formatMessage } = useIntl();
  const {
    selectedIndex,
    setSelectedIndex,
    onSubmit,
  } = useTranslationModal({ selectedLanguage, setSelectedLanguage, close });

  return (
    <StandardModal
      title={formatMessage(messages.languageSelectionModalTitle)}
      isOpen={isOpen}
      onClose={close}
      footerNode={(
        <ActionRow>
          <ActionRow.Spacer />
          <Button variant="tertiary" onClick={close}>
            {formatMessage(messages.cancelButtonText)}
          </Button>
          <Button onClick={onSubmit}>{formatMessage(messages.submitButtonText)}</Button>
        </ActionRow>
      )}
    >
      <ListBox className="listbox-container">
        {languages.map(([key, value], index) => (
          <ListBoxOption
            className="d-flex justify-content-between"
            key={key}
            selectedOptionIndex={selectedIndex}
            onSelect={() => setSelectedIndex(index)}
          >
            {formatMessage(languageMessages[value])}
            {selectedIndex === index && <Icon src={Check} />}
          </ListBoxOption>
        ))}
      </ListBox>
    </StandardModal>
  );
};

TranslationModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  close: PropTypes.func.isRequired,
  selectedLanguage: PropTypes.string.isRequired,
  setSelectedLanguage: PropTypes.func.isRequired,
};

export default TranslationModal;
