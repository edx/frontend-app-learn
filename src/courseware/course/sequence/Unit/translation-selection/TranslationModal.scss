.listbox-container {
  max-height: 400px;

  :last-child {
    margin-bottom: 5px;
  }
}