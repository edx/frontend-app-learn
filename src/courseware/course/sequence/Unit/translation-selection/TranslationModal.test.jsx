import { shallow } from '@edx/react-unit-test-utils';

import TranslationModal from './TranslationModal';

jest.mock('./useTranslationModal', () => ({
  __esModule: true,
  default: () => ({
    selectedIndex: 0,
    setSelectedIndex: jest.fn(),
    onSubmit: jest.fn().mockName('onSubmit'),
  }),
}));
jest.mock('@edx/paragon', () => jest.requireActual('@edx/react-unit-test-utils').mockComponents({
  StandardModal: 'StandardModal',
  ActionRow: {
    Spacer: 'Spacer',
  },
  Button: 'Button',
  Icon: 'Icon',
  ListBox: 'ListBox',
  ListBoxOption: 'ListBoxOption',
}));
jest.mock('@edx/paragon/icons', () => ({
  Check: jest.fn().mockName('icons.Check'),
}));
jest.mock('@edx/frontend-platform/i18n', () => {
  const i18n = jest.requireActual('@edx/frontend-platform/i18n');
  const { formatMessage } = jest.requireActual('@edx/react-unit-test-utils');
  // this provide consistent for the test on different platform/timezone
  const formatDate = jest.fn(date => new Date(date).toISOString()).mockName('useIntl.formatDate');
  return {
    ...i18n,
    useIntl: jest.fn(() => ({
      formatMessage,
      formatDate,
    })),
    defineMessages: m => m,
    FormattedMessage: () => 'FormattedMessage',
  };
});

describe('TranslationModal', () => {
  const props = {
    isOpen: true,
    close: jest.fn().mockName('close'),
    selectedLanguage: 'en',
    setSelectedLanguage: jest.fn().mockName('setSelectedLanguage'),
  };
  it('renders correctly', () => {
    const wrapper = shallow(<TranslationModal {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });
});
