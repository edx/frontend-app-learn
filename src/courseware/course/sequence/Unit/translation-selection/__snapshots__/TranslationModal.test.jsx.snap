// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslationModal renders correctly 1`] = `
<StandardModal
  footerNode={
    <ActionRow>
      <Spacer />
      <Button
        onClick={[MockFunction close]}
        variant="tertiary"
      >
        Cancel
      </Button>
      <Button
        onClick={[MockFunction onSubmit]}
      >
        Submit
      </Button>
    </ActionRow>
  }
  isOpen={true}
  onClose={[MockFunction close]}
  title="Translate this course"
>
  <ListBox
    className="listbox-container"
  >
    <ListBoxOption
      className="d-flex justify-content-between"
      key="en"
      onSelect={[Function]}
      selectedOptionIndex={0}
    >
      English
      <Icon
        src={[MockFunction icons.Check]}
      />
    </ListBoxOption>
    <ListBoxOption
      className="d-flex justify-content-between"
      key="es"
      onSelect={[Function]}
      selectedOptionIndex={0}
    >
      Spanish
    </ListBoxOption>
  </ListBox>
</StandardModal>
`;
